core/src/main/java/org/sonarsource/sonarlint/core/rules/RuleDetailsAdapter.java

```diff
@@ -123,22 +121,22 @@ private static Collection<RuleDescriptionTabDto> transformSectionsButIntroductio
       if (sectionsByKey.containsKey(sectionKey)) {
         var tabContents = sectionsByKey.get(sectionKey);
         Either<RuleNonContextualSectionDto, RuleContextualSectionWithDefaultContextKeyDto> content;
-        var matchingContext = tabContents.stream().filter(c -> c.getContext().isPresent() && c.getContext().get().getKey().equals(contextKey)).findFirst();
+        var foundMatchingContext = tabContents.stream().anyMatch(c -> c.getContext().isPresent() && c.getContext().get().getKey().equals(contextKey));
```

Comment:
_nitpick_: we could move the calculation of `foundMatchingContext ` just before line 139 when it's needed.